src/app/catalog/catalog.model.ts

```diff
@@ -18,3 +18,18 @@ export interface Size {
   id: number;
   size: string;
 }
+
+export interface Category {
+  id: number;
+  name: string;
+}
+
+export interface SubCategory {
+  id: number;
+  name: string;
+}
+
+export interface Brand {
+  id: number;
+  name: string;
```

Comment:
Instead of defining id and name for each interface, we can create a common interface with these props and extends it in these three..